src/server/set_family.cc

```diff
@@ -209,28 +209,47 @@ void InitSet(ArgSlice vals, CompactObj* set) {
   }
 }
 
-void ScanCallback(void* privdata, const dictEntry* de) {
-  StringVec* sv = (StringVec*)privdata;
-  sds key = (sds)de->key;
-  sv->push_back(string(key, sdslen(key)));
-}
-
 uint64_t ScanStrSet(const DbContext& db_context, const CompactObj& co, uint64_t curs,
-                    unsigned count, StringVec* res) {
+                    const ScanOpts& scan_op, StringVec* res) {
+  uint32_t count = scan_op.limit;
   long maxiterations = count * 10;
 
   if (IsDenseEncoding(co)) {
     StringSet* set = (StringSet*)co.RObjPtr();
     set->set_time(TimeNowSecRel(db_context.time_now_ms));
 
     do {
-      curs = set->Scan(curs, [&](const sds ptr) { res->push_back(std::string(ptr, sdslen(ptr))); });
+      auto scan_callback = [&](const sds ptr) {
+        sds s = (sds)ptr;
```

Comment:
do not think you need `s` - you can just use `ptr` below. Also, if you created `str`, then  `push_back(string(str))`